lib/platform/bitbucket-server/index.ts

```diff
@@ -891,12 +903,21 @@ export async function updatePr({
       updatePrVersion(pr.number, updatedStatePr.version);
     }
   } catch (err) {
+    logger.fatal({ err }, `Failed to update PR`);
     if (err.statusCode === 404) {
       throw new Error(REPOSITORY_NOT_FOUND);
     } else if (err.statusCode === 409) {
+      const errors = err?.response?.body?.errors || [];
+      if (
+        errors.every(
+          (error) => error?.exceptionName === BITBUCKET_INVALID_REVIEWERS
+        )
+      ) {
```

Comment:
We shouldn't duplicate this logic. Maybe better to introduce a new error code and throw it from below. Also, should we try re-submitting or re-updating the PR *without* reviewers?

Reply:
I extracted the logic into a separate function.

I also added the retry logic for `updatePR()`. I did sneak in an "internal" property in the object which is supplied to `updatePR`, so I can simply use recursion. Not sure if you are okay with doing this?

I did not add the retry logic to `addReviewers` yet, because of two things:

- The logic is different because imho it should not retry if the supplied username is invalid, but only if the username is one that is coming from the bb API. Unless you disagree I think it would be weird calling `addReviewers(5, ['name'])` whereas `name` is invalid and gets simply ignored.
- `addReviewers` has two fixed arguments so I'm not sure if it would be okay here to do the recursion by adding a 3rd optional argument like `invalidReviewers = null`. I guess that is somehow not okay as all the exported methods somehow follow some internal API for platforms?